app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -70,26 +79,120 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
+                    executor = txDetails.executor
+                )
+                txDetailsTransferBinding.created.value = txDetails.createdAt?.formatBackendDate() ?: ""
+
+                txDetailsTransferBinding.executed.value = txDetails.executedAt?.formatBackendDate() ?: ""
+                txDetailsTransferBinding.etherscan.setOnClickListener {
+                    requireContext().openUrl(
+                        getString(
+                            R.string.etherscan_transaction_url,
+                            txDetails.txHash
+                        )
                     )
+                }
+                val txType = if (txDetails.incoming == true) TxStatusView.TxType.TRANSFER_INCOMING else TxStatusView.TxType.TRANSFER_OUTGOING
+                txDetailsTransferBinding.txStatus.setStatus(txType, txDetails.txStatus)
+
+                if (txDetails.txData != null) {
+                    txDetailsTransferBinding.txData.visible(true)
+                    txDetails.txData?.hexData?.let { txDetailsTransferBinding.txData.setData(it, hexStringByteSize(it)) } // TODO: calculate right size or pass it through
+                } else {
+                    txDetailsTransferBinding.txData.visible(false)
+                    txDetailsTransferBinding.txDataSeparator.visible(false)
+                }
+            }
+            is SettingsChangeDetails -> {
+                val viewStub = binding.stubSettingsChange
+                if (viewStub.parent != null) {
+                    contentBinding = TxDetailsSettingsChangeBinding.bind(viewStub.inflate())
+                }
+                val txDetailsSettingsChangeBinding = contentBinding as TxDetailsSettingsChangeBinding
+
+                txDetailsSettingsChangeBinding.txConfirmations.setExecutionData(
+                    status = txDetails.txStatus,
+                    confirmations = (txDetails.detailedExecutionInfo as? MultisigExecutionDetails)?.confirmations?.map { it.signer } ?: listOf(),
+                    threshold = (txDetails.detailedExecutionInfo as? MultisigExecutionDetails)?.confirmationsRequired ?: 0,
+                    executor = txDetails.executor
                 )
+                txDetailsSettingsChangeBinding.created.value = txDetails.createdAt?.formatBackendDate() ?: ""
+                txDetailsSettingsChangeBinding.executed.value = txDetails.executedAt?.formatBackendDate() ?: ""
+
+                txDetailsSettingsChangeBinding.etherscan.setOnClickListener {
+                    requireContext().openUrl(
+                        getString(
+                            R.string.etherscan_transaction_url,
+                            txDetails.txHash
+                        )
+                    )
+                }
+                txDetailsSettingsChangeBinding.txStatus.setStatus(TxStatusView.TxType.MODIFY_SETTINGS, txDetails.txStatus)
+                if (txDetails.txData != null) {
```

Comment:
This section should also not show the txData section (see https://zpl.io/VYgpw7k)